### PR TITLE
New syntax for `dotnet new`

### PR DESCRIPTION
**Pull request by vlada-shubina**
## Summary
- changed `dotnet new` articles headers
- added information about new `dotnet new` syntax available from .NET SDK 7.0.100 Preview 2

### CODE REVIEW

**Comment by vlada-shubina** [comment]
@DamianEdwards do you want to update https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new-sdk-templates to include `--use-program-main` flag? (as in https://github.com/dotnet/docs/pull/28857/commits/3c742fcda94c179e6e09a0ddfb3e8e50cdfbf38c)

File: includes/dotnet-new-7-0-syntax.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+## .NET SDK 7.0.100 `dotnet new` syntax
+
+Starting .NET SDK 7.0.100  Preview 2 `dotnet new` syntax has changed:
+
+- `--list`, `--search`, `--install` `--uninstall` options became `list`, `search`, `install` and `uninstall` subcommand.
+- `--update-apply` option became `update` subcommand.
+- to use `--update-check`, use `update` subcommand with `--check` option.
+
+Other options available before are still available to use with respective subcommands.
```
Comment (baronfel) [inline includes/dotnet-new-7-0-syntax.md:16]:
```suggestion
Other options available before are still available to use with their respective subcommands.
```

**Review by baronfel — APPROVED** [review]
This looks great :) A couple small grammar suggestions are all I had.

**Comment by vlada-shubina** [comment]
@tdykstra i'm not sure about current `include`, would it be better to include it as a note, so syntax differs from main article content? Is there a similar syntax for docs?

File: docs/core/tools/dotnet-new-list.md [hunk 2]
```diff
@@ -22,6 +22,23 @@ dotnet new [<TEMPLATE_NAME>] -l|--list [--author <AUTHOR>] [-lang|--language {"C
...
 
+<!-- markdownlint-disable MD012 -->
+[!INCLUDE [new syntax](../../../includes/dotnet-new-7-0-syntax.md)]
+
+Examples:
+
+- Show help for list subcommand
+
+```
+dotnet new list --help
```
Comment (baronfel) [inline docs/core/tools/dotnet-new-list.md:33]: can you update [this example](https://github.com/dotnet/docs/blob/9afd7ac45aa807749e4bc07938c3c2b1e13f0c35/docs/core/tools/dotnet-new-sdk-templates.md?plain=1#L13) on the sdk templates page as well?

File: includes/dotnet-new-7-0-syntax.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ms.author: vshubina
+ms.date: 30/03/2022
+ms.topic: include
+---
+
+## .NET SDK 7.0.100 `dotnet new` syntax
+
+Starting .NET SDK 7.0.100  Preview 2 `dotnet new` syntax has changed:
+
+- `--list`, `--search`, `--install` `--uninstall` options became `list`, `search`, `install` and `uninstall` subcommand.
```
Comment (tdykstra) [inline includes/dotnet-new-7-0-syntax.md:12]:
```suggestion
- The `--list`, `--search`, `--install`, and `--uninstall` options became `list`, `search`, `install` and `uninstall` subcommands.
```

File: includes/dotnet-new-7-0-syntax.md (hunk 1 above)
Comment (tdykstra) [inline includes/dotnet-new-7-0-syntax.md]:
```suggestion
Other options that were available before are still available to use with their respective subcommands.
```

File: includes/dotnet-new-7-0-syntax.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+- `--update-apply` option became `update` subcommand.
+- to use `--update-check`, use `update` subcommand with `--check` option.
+
+Other options available before are still available to use with respective subcommands.
+There is a separate help for each subcommand avilable via `-h` or `--help` option: `dotnet new <subcommand> --help`, that lists all supported options for the subcommand.
```
Comment (tdykstra) [inline includes/dotnet-new-7-0-syntax.md:17]:
```suggestion
Separate help for each subcommand is available via the `-h` or `--help` option: `dotnet new <subcommand> --help` lists all supported options for the subcommand.
```

File: includes/dotnet-new-7-0-syntax.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+Additionally, the tab completion is now available for `dotnet new`.
```
Comment (tdykstra) [inline includes/dotnet-new-7-0-syntax.md]:
```suggestion
Additionally, tab completion is now available for `dotnet new`.
```

File: includes/dotnet-new-7-0-syntax.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+To activate tab completion for .NET SDK, follow [the guideline](../docs/core/tools/enable-tab-autocomplete.md).
```
Comment (tdykstra) [inline includes/dotnet-new-7-0-syntax.md:20]:
```suggestion
To activate tab completion for the .NET SDK, see [Enable tab completion](../docs/core/tools/enable-tab-autocomplete.md).
```

File: docs/core/tools/dotnet-new-install.md [hunk 7]
```diff
@@ -31,6 +31,23 @@ For example, if you install the template package using `dotnet new --install` in
 
 Starting with .NET SDK 6.0.100, installed template packages will available in later .NET SDK versions installed on your machine. The template package installed in .NET SDK 6.0.100 will also be available in .NET SDK 6.0.101, .NET SDK 6.0.200 and so on. However these template packages won't be available in .NET SDK versions prior to .NET SDK 6.0.100. To use template package installed .NET SDK 6.0.100 or later  in earlier .NET SDK versions, you need to install it using `dotnet new --install` using that .NET SDK version.
 
+<!-- markdownlint-disable MD012 -->
+[!INCLUDE [new syntax](../../../includes/dotnet-new-7-0-syntax.md)]
+
+Examples:
+
+- Show help for install subcommand
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-install.md:39]:
```suggestion
- Show help for `install` subcommand
```

File: docs/core/tools/dotnet-new-list.md [hunk 8]
```diff
@@ -22,6 +22,23 @@ dotnet new [<TEMPLATE_NAME>] -l|--list [--author <AUTHOR>] [-lang|--language {"C
 
 The `dotnet new --list` option lists available templates to use with `dotnet new`. If the <TEMPLATE_NAME> is specified, lists templates containing the specified name. This option lists only default and installed templates. To find templates in NuGet that you can install locally, use the [`--search`](dotnet-new-search.md) option.
 
+<!-- markdownlint-disable MD012 -->
+[!INCLUDE [new syntax](../../../includes/dotnet-new-7-0-syntax.md)]
+
+Examples:
+
+- Show help for list subcommand
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-list.md:30]:
```suggestion
- Show help for `list` subcommand
```

File: docs/core/tools/dotnet-new-search.md [hunk 9]
```diff
@@ -25,6 +25,24 @@ dotnet new [<TEMPLATE_NAME>] --search [--author <AUTHOR>] [-lang|--language {"C#
 
 The `dotnet new --search` option searches for templates supported by `dotnet new` on NuGet.org. When the <TEMPLATE_NAME> is specified, searches for templates containing the specified name.
 
+<!-- markdownlint-disable MD012 -->
+[!INCLUDE [new syntax](../../../includes/dotnet-new-7-0-syntax.md)]
+
+Examples:
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-search.md:31]:
```suggestion
Examples of the new syntax:
```

File: docs/core/tools/dotnet-new-search.md [hunk 10]
```diff
@@ -25,6 +25,24 @@ dotnet new [<TEMPLATE_NAME>] --search [--author <AUTHOR>] [-lang|--language {"C#
...
+
+- Show help for search subcommand
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-search.md:33]:
```suggestion
- Show help for the `search` subcommand.
```

File: docs/core/tools/dotnet-new-search.md [hunk 11]
```diff
@@ -25,6 +25,24 @@ dotnet new [<TEMPLATE_NAME>] --search [--author <AUTHOR>] [-lang|--language {"C#
...
+
+```
+dotnet new search --help
+```
+
+- Search for all templates available on NuGet.org matching the we substring and supporting the F# language
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-search.md:39]:
```suggestion
- Search for all templates available on NuGet.org matching the "we" substring and supporting the F# language
```

File: docs/core/tools/dotnet-new-uninstall.md [hunk 12]
```diff
@@ -21,6 +21,30 @@ dotnet new --uninstall <PATH|NUGET_ID>
 
 The `dotnet new --uninstall` command uninstalls a template package at the `PATH` or `NUGET_ID` provided. When the `<PATH|NUGET_ID>` value isn't specified, all currently installed template packages and their associated templates are displayed. When specifying `NUGET_ID`, don't include the version number.
 
+<!-- markdownlint-disable MD012 -->
+[!INCLUDE [new syntax](../../../includes/dotnet-new-7-0-syntax.md)]
+
+Examples:
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-uninstall.md:27]:
```suggestion
Examples of the new syntax:
```

File: docs/core/tools/dotnet-new-search.md [hunk 13]
```diff
@@ -25,6 +25,24 @@ dotnet new [<TEMPLATE_NAME>] --search [--author <AUTHOR>] [-lang|--language {"C#
...
+
+```
+dotnet new search we --language "F#"
+```
+
+
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-search.md:45]:
```suggestion
```

File: docs/core/tools/dotnet-new-uninstall.md [hunk 14]
```diff
@@ -21,6 +21,30 @@ dotnet new --uninstall <PATH|NUGET_ID>
...
+
+- Show help for uninstall subcommand
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-uninstall.md:29]:
```suggestion
- Show help for the `uninstall` subcommand.
```

File: docs/core/tools/dotnet-new-uninstall.md [hunk 15]
```diff
@@ -21,6 +21,30 @@ dotnet new --uninstall <PATH|NUGET_ID>
...
+dotnet new uninstall
+```
+
+- Uninstall the Azure web jobs project template package:
+
+```
+dotnet new uninstall Microsoft.Azure.WebJobs.ProjectTemplates
+```
+
+
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-uninstall.md:47]:
```suggestion
```

File: docs/core/tools/dotnet-new-update.md [hunk 16]
```diff
@@ -26,6 +26,31 @@ dotnet new --update-apply
 The `dotnet new --update-check` option checks if there are updates available for the template packages that are currently installed.
 The `dotnet new --update-apply` option checks if there are updates available for the template packages that are currently installed and installs them.
 
+<!-- markdownlint-disable MD012 -->
+[!INCLUDE [new syntax](../../../includes/dotnet-new-7-0-syntax.md)]
+
+Examples:
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-update.md:32]:
```suggestion
Examples of the new syntax:
```

File: docs/core/tools/dotnet-new-update.md [hunk 17]
```diff
@@ -26,6 +26,31 @@ dotnet new --update-apply
...
+
+- Show help for update subcommand
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-update.md:34]:
```suggestion
- Show help for the `update` subcommand.
```

File: docs/core/tools/dotnet-new-install.md [hunk 18]
```diff
@@ -31,6 +31,23 @@ For example, if you install the template package using `dotnet new --install` in
...
+
+```
+dotnet new install --help
```
Comment (tdykstra) [inline docs/core/tools/dotnet-new-install.md:42]: everywhere there's a block like this introduced by a bullet point, the block should be indented two spaces so the rendered example text will appear at the same indentation as the bullet text.

**Review by tdykstra — APPROVED** [review]
Looks great, I just made some grammar and formatting suggestions.

**Comment by tdykstra** [comment]
> @tdykstra i'm not sure about current `include`, would it be better to include it as a note, so syntax differs from main article content? Is there a similar syntax for docs?

That's a good idea, but then the new-syntax examples should be in the same format.  Syntax is here: https://review.docs.microsoft.com/en-us/help/contribute/markdown-reference?branch=main#alerts-note-tip-important-caution-warning

File: docs/core/tools/dotnet-new-install.md (hunk 18 above)
Comment (vlada-shubina) [reply docs/core/tools/dotnet-new-install.md:42]: fixed

File: docs/core/tools/dotnet-new-list.md (hunk 2 above)
Comment (vlada-shubina) [reply docs/core/tools/dotnet-new-list.md]: We decided not to update the other articles yet. This command will still work.
I haven't added a note to this article however, do you think we should add it here as well?

**Comment by vlada-shubina** [comment]
> added information about new console option --use-program-main available from .NET SDK 6.0.300

This edit was moved to https://github.com/dotnet/docs/pull/28858 together with web templates changes.

File: docs/core/tools/dotnet-new-list.md (hunk 2 above)
Comment (tdykstra) [reply docs/core/tools/dotnet-new-list.md]: I think it would be good to add it here as well.

**Comment by tdykstra** [comment]
> @tdykstra @baronfel I changed the paragraph to be a note, and also added it to `dotnet new <TEMPLATE>` article. Please check it, I prefer note style.

It makes sense to render it as a note. But it looks like the examples indentation was fixed only for `dotnet new list`.

**Comment by vlada-shubina** [comment]
> > @tdykstra @baronfel I changed the paragraph to be a note, and also added it to `dotnet new <TEMPLATE>` article. Please check it, I prefer note style.
> 
> It makes sense to render it as a note. But it looks like the examples indentation was fixed only for `dotnet new list`.

It looks i missed an extra space in note, thanks for catching - will fix tomorrow.

**Comment by vlada-shubina** [comment]
> > > @tdykstra I changed the paragraph to be a note, and also added it to `dotnet new <TEMPLATE>` article. Please check it, I prefer note style.
> > 
> > 
> > It makes sense to render it as a note. But it looks like the examples indentation was fixed only for `dotnet new list`.
> 
> It looks i missed an extra space in note, thanks for catching - will fix tomorrow.

Done - all pages have correct indentation now.

**Pull request merged**